File: app/src/main/java/uk/co/agfsoft/skyweather/volley/requests/GetByIdRequest.java
package uk.co.agfsoft.skyweather.volley.requests;

import com.android.volley.Request;

import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;

import uk.co.agfsoft.skyweather.BuildConfig;
import uk.co.agfsoft.skyweather.model.RawWeatherCity;
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest;



public class GetByIdRequest extends BaseVolleyRequest<RawWeatherCity> {

    private static final String ID = "id";
    private static final String APPID = "APPID";

    private Map<String, String> queryParams;

    @Inject
    public GetByIdRequest() {
    }

    @Override
    protected String getUrl() {
        return BuildConfig.OPEN_WEATHER_DATA_URL;
    }

    @Override
    protected int getMethod() {
        return Request.Method.GET;
    }

    @Override
    protected Class getResponseClass() {
        return RawWeatherCity.class;
    }


    @Override
    protected Map<String, String> getQueryParams() {
        return queryParams;
    }

    public void setQuery(int cityId) {
        queryParams = new HashMap<>(3);
        queryParams.put(ID, String.valueOf(cityId));
        queryParams.put(APPID, BuildConfig.OPEN_WEATHER_API_KEY);
    }
}

File: app/src/main/kotlin/volley/requests/GetByIdRequest.kt
package uk.co.agfsoft.skyweather.volley.requests

import com.android.volley.Request

import java.util.HashMap

import javax.inject.Inject

import uk.co.agfsoft.skyweather.BuildConfig
import model.RawWeatherCity
import uk.co.agfsoft.skyweather.volley.BaseVolleyRequest


class GetByIdRequest @Inject
constructor() : BaseVolleyRequest<RawWeatherCity>() {

    private var queryParams: MutableMap<String, String>? = null

    protected override val method: Int
        get() = Request.Method.GET

    protected override val responseClass: Class<*>
        get() = RawWeatherCity::class.java

    override fun getUrl(): String {
        return BuildConfig.OPEN_WEATHER_DATA_URL
    }


    override fun getQueryParams(): Map<String, String>? {
        return queryParams
    }

    fun setQuery(cityId: Int) {
        queryParams = HashMap(3)
        queryParams!![ID] = cityId.toString()
        queryParams!![APPID] = BuildConfig.OPEN_WEATHER_API_KEY
    }

    companion object {

        private val ID = "id"
        private val APPID = "APPID"
    }
}
